src/common/buffer.cc

```diff
@@ -540,6 +540,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
       if (r < 0) {
 	bdout << "raw_pipe: error setting nonblocking flag on temp pipe: "
 	      << cpp_strerror(r) << bendl;
+	close_pipe(tmpfd);
```

Comment:
might want to consolidate all the "close_pipe()` calls in the exit path to a single one with https://github.com/ceph/ceph/blob/master/src/include/scope_guard.h .